src/space_usage.rs

```diff
@@ -0,0 +1,110 @@
+//! Module to collect and display Disk Space Usage of a Profile.
+use crate::context::Context;
+use anyhow::Result;
+use humansize::{BINARY, format_size};
+
+/// Space Usage Report
+/// Useful for debugging space usage problems in the deltachat database.
+#[derive(Debug)]
+pub struct SpaceUsage {
+    /// Total database size, subtract this from the backup size to estimate size of all blobs
+    pub db_size: usize,
+    /// size and row count of the 10 biggest tables
+    pub largest_tables: Vec<(String, usize, Option<usize>)>,
+    /// count and total size of status updates
+    /// for the 10 webxdc apps with the most size usage in status updates
+    pub largest_webxdc_data: Vec<(usize, usize, usize)>,
+}
+
+impl std::fmt::Display for SpaceUsage {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "SpaceUsage:")?;
+        let human_db_size = format_size(self.db_size, BINARY);
+        writeln!(f, "[Database Size]: {human_db_size}")?;
+        writeln!(f, "[Largest Tables]:")?;
+        for (name, size, row_count) in &self.largest_tables {
+            let human_table_size = format_size(*size, BINARY);
+            writeln!(
+                f,
+                "   {name:<20} {human_table_size:>10}, {row_count:>6} rows",
+                name = format!("{name}:"),
+                row_count = row_count.map(|c| c.to_string()).unwrap_or("?".to_owned())
+            )?;
+        }
+        writeln!(f, "[Webxdc With Biggest Status Update Space Usage]:")?;
+        for (msg_id, size, update_count) in &self.largest_webxdc_data {
+            let human_size = format_size(*size, BINARY);
+            writeln!(
+                f,
+                "   {msg_id:<8} {human_size:>10} across {update_count:>5} updates",
+                msg_id = format!("{msg_id}:")
+            )?;
+        }
+        Ok(())
+    }
+}
+
+/// Get space usage information for the Context's database
+pub async fn get_space_usage(ctx: &Context) -> Result<SpaceUsage> {
+    let page_size: usize = ctx
+        .sql
+        .query_get_value("PRAGMA page_size", ())
+        .await?
+        .unwrap_or_default();
+    let page_count: usize = ctx
+        .sql
+        .query_get_value("PRAGMA page_count", ())
+        .await?
+        .unwrap_or_default();
+
+    let mut largest_tables = Vec::new();
+
+    let biggest_tables = ctx
```

Comment:
`larges_tables` is `mut` anyway, so we could fill it right away, but with row counts = 0, and then update row counts in place, to get rid of `biggest_tables` var. It's not clear how "largest" differs from "biggest" when looking at the code at first